pallets/multicollateral-bonding-curve-pool/src/lib.rs

```diff
@@ -1588,6 +1589,59 @@ impl<T: Config> LiquiditySource<T::DEXId, T::AccountId, T::AssetId, Balance, Dis
         }
     }
 
+    fn step_quote(
+        dex_id: &T::DEXId,
+        input_asset_id: &T::AssetId,
+        output_asset_id: &T::AssetId,
+        amount: QuoteAmount<Balance>,
+        samples_count: usize,
+        deduce_fee: bool,
+    ) -> Result<VecDeque<SwapChunk<Balance>>, DispatchError> {
+        if !Self::can_exchange(dex_id, input_asset_id, output_asset_id) {
+            fail!(Error::<T>::CantExchange);
+        }
+        if amount.amount().is_zero() {
+            return Ok(VecDeque::new());
+        }
+
+        let base_asset_id = &T::GetBaseAssetId::get();
+
+        let step = amount
+            .amount()
+            .checked_div(samples_count as Balance)
+            .ok_or(Error::<T>::ArithmeticError)?;
+
+        let mut chunks = VecDeque::new();
+        let mut sub_in = Balance::zero();
+        let mut sub_out = Balance::zero();
+        let mut sub_fee = Balance::zero();
+
+        for i in 1..=samples_count {
+            let volume = amount.copy_direction(
+                step.checked_mul(i as Balance)
+                    .ok_or(Error::<T>::ArithmeticError)?,
+            );
```

Comment:
If I'm not mistaken, when `i = samples_count`, `volume` might be smaller than `amount` (because of rounding on `checked_div`), so the sum of chunks is smaller than requested `amount`. Not sure about use cases of `step_quote` and whether this might be a problem.

Reply:
Yes, it is acceptable rounding that will be covered by slippage